src/net/sourceforge/kolmafia/Modifiers.java

```diff
@@ -1236,32 +1238,49 @@ public static final String getDerivedModifierName(final int index) {
     return Modifiers.modifierName(Modifiers.derivedModifiers, index);
   }
 
-  private static String modifierName(final Object[][] table, final int index) {
+  private static <T> T modifierRow(
+      final Object[][] table, final int index, Function<Object[], T> getter) {
     if (index < 0 || index >= table.length) {
       return null;
     }
-    return (String) table[index][0];
+
+    return getter.apply(table[index]);
   }
 
-  private static Object modifierDescPattern(final Object[][] table, final int index) {
-    if (index < 0 || index >= table.length) {
-      return null;
-    }
-    return table[index][1];
+  private static String modifierName(final Object[][] table, final int index) {
+    return modifierRow(table, index, Modifiers::modifierName);
+  }
+
+  private static String modifierName(final Object[] tableRow) {
+    return (String) tableRow[0];
+  }
+
+  private static Pattern[] modifierDescPattern(final Object[][] table, final int index) {
+    return modifierRow(table, index, Modifiers::modifierDescPattern);
```

Comment:
Please ensure this file is tested more thoroughly.

I know it wasn't tested before, but this is some extensive refactoring.

Reply:
I've added loads of tests to it